# Spotify Top 50 Musics in 2025 - Exploração inicial

### 1. Carregando o dataset

In [1]:
import pandas as pd
#index_col=0 impede que o pandas crie uma coluna extra de índice
df = pd.read_csv('2025_top50_songs.csv', index_col=0)
df.head()

,song_title,artist,streams_2025_billions,primary_genre,bpm,duration_seconds,release_year,artist_country,explicit,danceability,energy,valence,acousticness,peak_global_chart_position,dataset_part
wrapped_2025_rank,,,,,,,,,,,,,,,
1,Die With A Smile,Lady Gaga & Bruno Mars,1.70,Pop,120,251,2024,USA,False,0.72,0.68,0.55,0.18,1,Spotify Wrapped 2025 Top 50 Global Songs
2,APT.,ROSE & Bruno Mars,1.62,K-Pop/Pop,145,173,2024,South Korea,False,0.86,0.79,0.88,0.05,1,Spotify Wrapped 2025 Top 50 Global Songs
3,Espresso,Sabrina Carpenter,1.55,Pop,104,175,2024,USA,False,0.80,0.73,0.96,0.06,1,Spotify Wrapped 2025 Top 50 Global Songs
4,Please Please Please,Sabrina Carpenter,1.48,Pop,107,186,2024,USA,False,0.70,0.66,0.87,0.10,1,Spotify Wrapped 2025 Top 50 Global Songs
5,Taste,Sabrina Carpenter,1.41,Pop,117,177,2024,USA,True,0.78,0.70,0.82,0.08,1,Spotify Wrapped 2025 Top 50 Global Songs


### 2. Dimensões e estrutura

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 50 entries, 1 to 50
Data columns (total 15 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   song_title                  50 non-null     object 
 1   artist                      50 non-null     object 
 2   streams_2025_billions       50 non-null     float64
 3   primary_genre               50 non-null     object 
 4   bpm                         50 non-null     int64  
 5   duration_seconds            50 non-null     int64  
 6   release_year                50 non-null     int64  
 7   artist_country              50 non-null     object 
 8   explicit                    50 non-null     bool   
 9   danceability                50 non-null     float64
 10  energy                      50 non-null     float64
 11  valence                     50 non-null     float64
 12  acousticness                50 non-null     float64
 13  peak_global_chart_position  50 non-null   

In [3]:
df.describe()

,streams_2025_billions,bpm,duration_seconds,release_year,danceability,energy,valence,acousticness,peak_global_chart_position
count,50.000000,50.000000,50.000000,50.00000,50.000000,50.000000,50.00000,50.000000,50.000000
mean,0.929400,109.800000,205.440000,2021.86000,0.681600,0.646800,0.66560,0.152200,1.960000
std,0.303022,23.363149,34.546721,6.54345,0.114898,0.120838,0.18306,0.123739,3.070332
min,0.560000,72.000000,138.000000,1985.00000,0.410000,0.400000,0.22000,0.000000,1.000000
25%,0.692500,92.250000,178.000000,2023.00000,0.600000,0.550000,0.55250,0.060000,1.000000
50%,0.845000,104.000000,198.000000,2024.00000,0.700000,0.670000,0.68000,0.100000,1.000000
75%,1.112500,119.500000,222.750000,2024.00000,0.777500,0.740000,0.80000,0.217500,2.000000
max,1.700000,170.000000,301.000000,2024.00000,0.860000,0.900000,0.96000,0.500000,22.000000


In [4]:
df.shape

(50, 15)

### 3. Valores nulos

In [5]:
nulos = df.isnull().sum()
nulos
#nenhum dado vazio

,0
song_title,0
artist,0
streams_2025_billions,0
primary_genre,0
bpm,0
duration_seconds,0
release_year,0
artist_country,0
explicit,0
danceability,0


### 4. Valores únicos

In [6]:
df['artist'].nunique()

35

In [7]:
df['artist_country'].nunique()

10

In [8]:
df['primary_genre'].nunique()

23

In [9]:
df['artist_country'].value_counts()

,count
artist_country,
USA,36
UK,3
South Korea,2
Colombia,2
South Africa,2
Ireland,1
Canada,1
Chile,1
Netherlands,1


In [10]:
df['artist'].value_counts()

,count
artist,
Sabrina Carpenter,4
Chappell Roan,3
Billie Eilish,3
Taylor Swift,3
Beyoncé,2
Olivia Rodrigo,2
Kendrick Lamar,2
Shaboozey,2
Gracie Abrams,2


In [11]:
df['primary_genre'].value_counts()

,count
primary_genre,
Pop,10
Indie Pop,7
Synth-Pop,4
Country,3
Pop Rock,2
Hip-Hop,2
K-Pop/Pop,2
Reggaeton,2
Afrobeats,2


### Conclusões

* O dataset contém dados das 50 músicas mais ouvidas em 2025 na plataforma **Spotify**, contendo dados como bpm das músicas, ano de lançamento e até valores como dançabilidade, duração da música em segundos. o df é formado por **50** linhas e **15** colunas, com nenhum dado nulo/vazio

  Usando o comando describe, conseguimos informações interessantes:

  **Médias**

  bpm: **109.8** - as músicas seguem um padrão de moderado a animado,
  abrangendo todo tipo de gosto musical.

  duration_seconds: **205.44** (aproximadamente 3 min 25 s) - um valor
  considerável levando em conta o imediatismo que a sociedade vive hoje.

  danceability: **0.681** - assim como o bpm, está na média. São músicas
  que você pode dançar ou simplesmente curtir a vibe.

* O dataset revela um domínio claro dos **EUA**, com **36 dos 50 artistas (72%)**
  — o mercado global de streaming ainda é amplamente americano. O **Reino Unido**
  aparece em segundo com apenas 3 artistas, seguido de **Coreia do Sul** e
  **Colômbia** com 2 cada.

* Em relação aos gêneros, o **Pop** lidera com 10 músicas, mas se somarmos
  todas as suas variações — Indie Pop, Synth-Pop, K-Pop/Pop, Pop Rock —
  o domínio é ainda maior, representando mais da metade do ranking.

* **Sabrina Carpenter** é a artista mais presente, com **4 músicas** no top 50
  — única a ultrapassar 3 aparições. Billie Eilish, Chappell Roan e Taylor Swift
  aparecem com 3 músicas cada, domínio absurdo de cantoras do sexo feminino.